src/D2L.Security.OAuth2/Keys/Default/MonoRsaSecurityKey.cs

```diff
@@ -28,23 +28,43 @@ internal sealed class MonoRsaSecurityKey : AsymmetricSecurityKey {
 		}
 
 		public override bool IsAsymmetricAlgorithm( string algorithm ) {
-			switch( algorithm ) {
+			if (String.IsNullOrWhiteSpace(algorithm)) {
+				throw new ArgumentNullException( "algorithm" );
+			}
+
+			switch ( algorithm ) {
 				case SecurityAlgorithms.RsaSha1Signature:
 				case SecurityAlgorithms.RsaSha256Signature:
 				case SecurityAlgorithms.RsaOaepKeyWrap:
 				case SecurityAlgorithms.RsaV15KeyWrap:
 					return true;
-				default:
-					return false;
 			}
+
+			return false;

```

Comment:
this should be `throw new NotSupportedException( $"{algorithm} is not supported" );